sdk/spring/azure-spring-cloud-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/config/AzureProperties.java

```diff
@@ -41,7 +41,7 @@ private void validate() {
         }
 
         if (msiEnabled && Strings.isNullOrEmpty(subscriptionId)) {
-            Assert.hasText(this.region, "When msi is enabled, spring.cloud.azure.subscription-id must be provided");
+            Assert.hasText(this.subscriptionId, "When msi is enabled, spring.cloud.azure.subscription-id must be provided");
```

Comment:
Can we delete ` && Strings.isNullOrEmpty(subscriptionId)`?

Reply:
> Can we delete ` && Strings.isNullOrEmpty(subscriptionId)`?

I think it's necessary, we will assert it when the subscriptionId is empty

Reply:
Here is the source code of `Assert.hasText(...)`:

```
	public static void hasText(@Nullable String text, String message) {
		if (!StringUtils.hasText(text)) {
			throw new IllegalArgumentException(message);
		}
	}
```

It will check by `StringUtils.hasText(text)`.

Maybe we do not need check it again by `Strings.isNullOrEmpty(subscriptionId)`?